File: backend/RL.API/Models/ListasModels.cs
using System;

namespace RL.API.DTOs
{
    public class CoincidenciaJuridicaDto
    {
        public string Rtn { get; set; } = string.Empty;
        public string Nombre { get; set; } = string.Empty;
        public string NumeroPatrono { get; set; } = string.Empty;
        public string ListaCoincidencia { get; set; } = string.Empty;
        public DateTime? FechaEncontro { get; set; }
        public DateTime? FechaCalifico { get; set; }
        public string EsProveedorIhss { get; set; } = string.Empty;
        public bool TieneMotivo { get; set; }
        public bool EsManual { get; set; }
    }

    public class CoincidenciaNaturalDto
    {
        public string NumeroIdentificacion { get; set; } = string.Empty;
        public string Nombre { get; set; } = string.Empty;
        public string ListaCoincidencia { get; set; } = string.Empty;
        public int TotalRepetidos { get; set; }
        public bool TieneMotivo { get; set; }
        public bool EsManual { get; set; }
    }

    public class CoincidenciaEmpleadoDto
    {
        public string Identidad { get; set; } = string.Empty;
        public string Nombre { get; set; } = string.Empty;
        public string ListaCoincidencia { get; set; } = string.Empty;
        public int TotalRepetidos { get; set; }
        public bool TieneMotivo { get; set; }
        public bool EsManual { get; set; }
    }

    public class DetalleCoincidenciaNaturalDto
    {
        public string NumeroIdentificacion { get; set; } = string.Empty;
        public string NombresPersona { get; set; } = string.Empty;
        public string TipoCondicionActuaDesc { get; set; } = string.Empty;
        public string NumeroPatronal { get; set; } = string.Empty;
        public string NombreEmpresa { get; set; } = string.Empty;
        public string EsPep { get; set; } = string.Empty;
        public string ListaCoincidencia { get; set; } = string.Empty;
        public DateTime? FechaCalifico { get; set; }
        public DateTime? FechaCoincidencia { get; set; }
    }

    public class DetalleCoincidenciaEmpleadoDto
    {
        public string Identidad { get; set; } = string.Empty;
        public string Nombre { get; set; } = string.Empty;
        public string TipoCondicionActuaDesc { get; set; } = string.Empty;
        public string NumeroPatrono { get; set; } = string.Empty;
        public string NombreEmpresa { get; set; } = string.Empty;
        public string RazoSoci { get; set; } = string.Empty;
        public string ListaCoincidencia { get; set; } = string.Empty;
        public DateTime? FechaCalifico { get; set; }
        public DateTime? FechaCoincidencia { get; set; }
    }
}

File: frontend/rl-app/src/app/features/admin/listas/models/listas.models.ts
export interface TipoDocumento {
  tipoDocumentoId: number;
  descripcion: string;
}

export interface TipoListaCautela {
  tipoListaCautelaId: number;
  descripcion: string;
  tipoArchivo?: string | null;
  cantidadColumnas?: number | null;
}

export interface ResumenLista {
  tipoListaCautelaId?: number;
  lista: string;
  usuario: string;
  fechaCreacion?: string;
  cantidadRegistros: number;
}

export interface RegistrarPositivoDto {
  tipoDocumentoId: number;
  tipoPositivoId: number;
  noDocumento: string;
  nombreCompleto: string;
  motivoIngreso: string;
  tipoListaCautelaId?: number | null;
  origenRegistro?: string | null;
}

export interface ExistingPositivo {
  tipoDocumentoId: number;
  motivoIngreso: string;
  tipoListaCautelaId?: number | null;
  origenRegistro?: string | null;
  fechaRegistroInterno?: string | null;
}

export interface EvidenciaPolitica {
  maximoMb: number;
  maximoBytes: number;
  extensionesPermitidas: string[];
  tiposPermitidosTexto: string;
}

export interface CoincidenciaJuridica {
  rtn: string;
  nombre: string;
  numeroPatrono: string;
  listaCoincidencia: string;
  fechaEncontro?: string;
  fechaCalifico?: string;
  fechaRegistroInterno?: string | null;
  esProveedorIhss?: string;
  tieneMotivo?: boolean;
  esManual?: boolean;
}

export interface CoincidenciaNatural {
  numeroIdentificacion: string;
  nombre: string;
  listaCoincidencia: string;
  totalRepetidos: number;
  fechaEncontro?: string | null;
  fechaCalifico?: string | null;
  fechaRegistroInterno?: string | null;
  tieneMotivo?: boolean;
  esManual?: boolean;
}

export interface CoincidenciaEmpleado {
  identidad: string;
  nombre: string;
  listaCoincidencia: string;
  totalRepetidos: number;
  fechaEncontro?: string | null;
  fechaCalifico?: string | null;
  fechaRegistroInterno?: string | null;
  tieneMotivo?: boolean;
  esManual?: boolean;
}

export interface DetalleCoincidenciaNatural {
  numeroIdentificacion: string;
  nombresPersona: string;
  tipoCondicionActuaDesc: string;
  numeroPatronal: string;
  nombreEmpresa: string;
  esPep: string;
  listaCoincidencia: string;
  fechaCalifico?: string;
  fechaCoincidencia?: string;
}

export interface DetalleCoincidenciaEmpleado {
  identidad: string;
  nombre: string;
  tipoCondicionActuaDesc: string;
  numeroPatrono: string;
  nombreEmpresa: string;
  razoSoci: string;
  listaCoincidencia: string;
  fechaCalifico?: string;
  fechaCoincidencia?: string;
}

export interface CoincidenciaPatronoResumen {
  fechaEncontro: string;
  cantidadRegistros: number;
}

export type CoincidenciaEmpleadoResumen = CoincidenciaPatronoResumen;

export interface CoincidenciaPatronoDetalle {
  reporteCoincidenciaId: number;
  dataId: number;
  dni: string;
  fechaEncontro: string;
  listaCoincidencia: string;
  nacionalidad: string;
  nombre: string;
  numeroPatrono: string;
  observacionLista: string;
  tipoPersona: string;
  usuarioEncontro: number;
  tipoCalificacion: string;
}

export type CoincidenciaEmpleadoDetalle = CoincidenciaPatronoDetalle;

export interface Evidencia {
  evidenciaId: number;
  nombreArchivo: string;
  tipoMime: string;
}

export interface Seguimiento {
  detalleListaId: number;
  positivoId: number;
  motivoIngreso: string;
  fechaCreacion: string;
  usrCreacionId: number;
  usrEmail: string;
  evidencias: Evidencia[];
}
